lib/IniProcessor/ini_processing.cpp

```diff
@@ -653,6 +653,43 @@ bool IniProcessing::hasKey(const std::string &keyName) const
     return (e != m_params.currentGroup->end());
 }
 
+bool IniProcessing::renameKey(const std::string &oldName, const std::string &newName)
+{
+    if(!m_params.opened)
+        return false;
+
+    if(!m_params.currentGroup)
+        return false;
+
+    auto it = m_params.currentGroup->find(oldName);
+    if(it != m_params.currentGroup->end())
+    {
+        std::swap((*m_params.currentGroup)[newName], it->second);
+        m_params.currentGroup->erase(it);
+        return true;
+    }
```

Comment:
What happen if old name and new name will be equal? :)
Looks suspicious...

Reply:
P.S. About case sensitivity, pay attention on this code: https://github.com/Wohlstand/TheXTech/blob/bd7de09274a49d221055a2edc3877934a5310712/lib/IniProcessor/ini_processing.h#L95